src/main/scala/sbtavrohugger/FileWriter.scala

```diff
@@ -11,37 +11,42 @@ import sbt.Path._
 object FileWriter {
 
   private[sbtavrohugger] def generateCaseClasses(
-    generator: Generator,
-    srcDirs: Seq[File],
-    target: File,
-    log: Logger): Set[java.io.File] = {
+                                                  generator: Generator,
+                                                  srcDirs: Seq[File],
+                                                  target: File,
+                                                  log: Logger): Set[java.io.File] = {
     log.info("Considering source directories %s".format(srcDirs.mkString(",")))
+
     def getSrcFiles(dirs: Seq[File], fileExtension: String) = for {
       srcDir <- dirs
       srcFile <- (srcDir ** s"*.$fileExtension").get
     } yield srcFile
-    
-    for (inFile <- AvscFileSorter.sortSchemaFiles(getSrcFiles(srcDirs, "avsc"))) {
```

Comment:
looking great. if we can restore the filesorting step, I think we'll be good to release.

>sbt:sbt-avrohugger> scripted avrohugger/filesorter
[info] Wrote /Users/julianpeeters/Workspace/sbt-avrohugger/target/scala-2.12/sbt-1.0/sbt-avrohugger-2.10.0-SNAPSHOT.pom
[info] Wrote /Users/julianpeeters/Workspace/sbt-avrohugger/target/scala-2.12/sbt-1.0/sbt-avrohugger_2.12_1.0-2.10.0-SNAPSHOT.pom
[info] :: delivering :: com.julianpeeters#sbt-avrohugger;2.10.0-SNAPSHOT :: 2.10.0-SNAPSHOT :: integration :: Thu Dec 26 12:34:36 PST 2024
[info]  delivering ivy file to /Users/julianpeeters/Workspace/sbt-avrohugger/target/scala-2.12/sbt-1.0/ivy-2.10.0-SNAPSHOT.xml
[info]  published sbt-avrohugger_2.12_1.0 to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/poms/sbt-avrohugger_2.12_1.0.pom
[info]  published sbt-avrohugger to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/docs/sbt-avrohugger-javadoc.jar
[info]  published sbt-avrohugger_2.12_1.0 to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/srcs/sbt-avrohugger_2.12_1.0-sources.jar
[info]  published sbt-avrohugger to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/jars/sbt-avrohugger.jar
[info]  published sbt-avrohugger_2.12_1.0 to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/docs/sbt-avrohugger_2.12_1.0-javadoc.jar
[info]  published sbt-avrohugger to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/poms/sbt-avrohugger.pom
[info]  published sbt-avrohugger to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/srcs/sbt-avrohugger-sources.jar
[info]  published sbt-avrohugger_2.12_1.0 to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/jars/sbt-avrohugger_2.12_1.0.jar
[info]  published ivy to /Users/julianpeeters/.ivy2/local/com.julianpeeters/sbt-avrohugger/scala_2.12/sbt_1.0/2.10.0-SNAPSHOT/ivys/ivy.xml
[info] Running 1 / 1 (100.00%) scripted tests with RunFromSourceMain
[info] Running avrohugger/filesorter
[error] [info] [launcher] getting org.scala-sbt sbt 1.10.6  (this may take some time)...
[error] [info] [launcher] getting Scala 2.12.20 (for sbt)...
[info] [info] welcome to sbt 1.10.6 (Eclipse Adoptium Java 21.0.5)
[info] [info] loading settings for project sbt_7543aab3-build from plugins.sbt...
[info] [info] loading project definition from /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/project
[info] [info] loading settings for project sbt_7543aab3 from build.sbt...
[info] [info] set current project to filesorter-test (in build file:/private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/)
[info] [info] welcome to sbt 1.10.6 (Eclipse Adoptium Java 21.0.5)
[info] [info] loading settings for project sbt_7543aab3-build from plugins.sbt...
[info] [info] loading project definition from /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/project
[info] [info] loading settings for project sbt_7543aab3 from build.sbt...
[info] [info] set current project to filesorter-test (in build file:/private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/)
[info] [info] Considering source directories /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro
[info] [info] Compiling AVSC files 
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_d.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/f.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_e.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_b.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/a.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_c.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/b.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_a.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/c.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/d.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/_f.avsc
[info] [info] /private/var/folders/1m/4vqhbccd6l7dkz3t8q766fdr0000gn/T/sbt_7543aab3/src/main/avro/e.avsc
[info] [error] org.apache.avro.SchemaParseException: "_C" is not a defined name. The type of the "_c" field must be a defined name or a {"type": ...} expression.
[info] [error]  at org.apache.avro.Schema.parse(Schema.java:1734)
[info] [error]  at org.apache.avro.Schema$Parser.parse(Schema.java:1471)
[info] [error]  at org.apache.avro.Schema$Parser.parse(Schema.java:1433)
[info] [error]  at avrohugger.input.parsers.FileInputParser$$anonfun$1.$anonfun$applyOrElse$1(FileInputParser.scala:72)
[info] [error]  at scala.util.Try$.apply(Try.scala:213)
[info] [error]  at avrohugger.input.parsers.FileInputParser$$anonfun$1.applyOrElse(FileInputParser.scala:72)
[info] [error]  at avrohugger.input.parsers.FileInputParser$$anonfun$1.applyOrElse(FileInputParser.scala:71)
[info] [error]  at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info] [error]  at scala.util.Failure.recoverWith(Try.scala:236)
[info] [error]  at avrohugger.input.parsers.FileInputParser.tryParse$1(FileInputParser.scala:71)
[info] [error]  at avrohugger.input.parsers.FileInputParser.getSchemaOrProtocols(FileInputParser.scala:86)
[info] [error]  at avrohugger.generators.FileGenerator.$anonfun$filesToFiles$1(FileGenerator.scala:99)
[info] [error]  at scala.collection.immutable.List.flatMap(List.scala:366)
[info] [error]  at avrohugger.generators.FileGenerator.filesToFiles(FileGenerator.scala:99)
[info] [error]  at avrohugger.Generator.filesToFiles(Generator.scala:98)
[info] [error]  at sbtavrohugger.FileWriter$.generateCaseClasses(FileWriter.scala:28)
[info] [error]  at sbtavrohugger.SbtAvrohugger$.$anonfun$avroSettings$8(SbtAvrohugger.scala:82)
[info] [error]  at sbt.util.FileFunction$.$anonfun$cached$1(FileFunction.scala:81)
[info] [error]  at sbt.util.FileFunction$.$anonfun$cached$4(FileFunction.scala:154)
[info] [error]  at sbt.util.Difference.apply(Tracked.scala:415)
[info] [error]  at sbt.util.Difference.apply(Tracked.scala:395)
[info] [error]  at sbt.util.FileFunction$.$anonfun$cached$3(FileFunction.scala:150)
[info] [error]  at sbt.util.Difference.apply(Tracked.scala:415)
[info] [error]  at sbt.util.Difference.apply(Tracked.scala:390)
[info] [error]  at sbt.util.FileFunction$.$anonfun$cached$2(FileFunction.scala:149)
[info] [error]  at sbtavrohugger.SbtAvrohugger$.$anonfun$avroSettings$7(SbtAvrohugger.scala:84)
[info] [error]  at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[info] [error]  at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[info] [error]  at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[info] [error]  at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[info] [error]  at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[info] [error]  at sbt.Execute.work(Execute.scala:292)
[info] [error]  at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[info] [error]  at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info] [error]  at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[info] [error]  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[info] [error]  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[info] [error]  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[info] [error]  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[info] [error]  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[info] [error]  at java.base/java.lang.Thread.run(Thread.java:1583)
[info] [error] (Compile / avroScalaGenerate) org.apache.avro.SchemaParseException: "_C" is not a defined name. The type of the "_c" field must be a defined name or a {"type": ...} expression.
[info] [error] Total time: 0 s, completed Dec 26, 2024, 12:34:41 PM
[error] x avrohugger/filesorter 
[error]  Cause of test exception: {line 1}  Command failed: compile failed
[error] stack trace is suppressed; run last scripted for the full output
[error] (scripted) Failed tests:
[error]         avrohugger/filesorter

Reply:
Indeed, my bad! Both PRs should be ok now.